test/kt/esimd_radix_sort_by_key.cpp

```diff
@@ -0,0 +1,128 @@
+// -*- C++ -*-
+//===-- esimd_radix_sort_by_key.cpp -----------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "../support/test_config.h"
+#include "../support/utils.h"
+#include "esimd_radix_sort_utils.h"
+
+#include <oneapi/dpl/experimental/kernel_templates>
+#include <oneapi/dpl/execution>
+#include <oneapi/dpl/algorithm>
+#include <oneapi/dpl/iterator>
+
+#include <vector>
+#include <string>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+
+template<typename KeyT, typename ValueT, bool isAscending, std::uint32_t RadixBits, typename KernelParam>
+void test_sycl_iterators(std::size_t size, KernelParam param)
+{
+    sycl::queue q = TestUtils::get_test_queue();
+
+    std::vector<KeyT> expected_keys(size);
+    std::vector<ValueT> expected_values(size);
+    generate_data(expected_keys.data(), size, 6);
+    generate_data(expected_values.data(), size, 7);
+
+    std::vector<KeyT> actual_keys(expected_keys);
+    std::vector<ValueT> actual_values(expected_values);
+    {
+        sycl::buffer<KeyT> keys(actual_keys.data(), actual_keys.size());
+        sycl::buffer<ValueT> values(actual_values.data(), actual_values.size());
+        oneapi::dpl::experimental::kt::esimd::radix_sort_by_key<isAscending, RadixBits>(
+            q, oneapi::dpl::begin(keys), oneapi::dpl::end(keys), oneapi::dpl::begin(values), param).wait();
+    }
+
+    auto expected_first  = oneapi::dpl::make_zip_iterator(std::begin(expected_keys), std::begin(expected_values));
+    std::stable_sort(expected_first, expected_first + size, CompareKey<isAscending>{});
+
+    std::string parameters_msg = ", n: " + std::to_string(size) +
```

Comment:
I believe better to use `std::stringstream` in this case and simple write to stream all required info through operator `<<`.
In this case we don't need make the calls like `std::to_string`.